core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -793,6 +822,18 @@ private static Map<String, SnapshotRef> validateRefs(
     return inputRefs;
   }
 
+  private long computeStatisticsCheckpointSnapshotID() {
```

Comment:
This doesn't make sense to me and I suspect it should be removed. There is no guaranteed order to `snapshots` and the snapshots in that list can be from different branches. The first snapshot with a valid stats file has no meaning.

Reply:
I didn't consider the branching. True. I will remove it.